smithy-model/src/main/java/software/amazon/smithy/model/loader/IdlModelParser.java

```diff
@@ -532,25 +541,33 @@ private void parseMember(ShapeId parent, Set<String> allowed, Set<String> define
         }
 
         ShapeId memberId = parent.withMember(memberName);
+
+        if (isTargetElided && !modelFile.getVersion().supportsTargetElision()) {
```

Comment:
why is this not right after detecting elision?

Reply:
Using the member id gives a much better error message